src/pages/add-lock-exception/AddLockExceptionForm/AddLockExceptionForm.js

```diff
@@ -0,0 +1,79 @@
+import i18n from '@dhis2/d2-i18n'
+import { SingleSelectField, SingleSelectOption } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import styles from './AddLockExceptionForm.module.css'
+import OrganisationUnitSelection from './OrganisationUnitSelection/OrganisationUnitSelection'
+import PeriodPicker from './PeriodPicker'
+
+const AddLockExceptionForm = ({
+    dataSets,
+    groups,
+    levels,
+    selectedDataSetId,
+    selectedOrgUnits,
+    onSelectedDataSetIdChange,
+    onSelectedOrgUnitsChange,
+    onSelectedPeriodIdChange,
+}) => {
+    const handleDataSetChange = ({ selected: dataSetId }) => {
+        if (selectedDataSetId !== null && dataSetId === selectedDataSetId) {
+            return
+        }
+        onSelectedDataSetIdChange(dataSetId)
+        onSelectedOrgUnitsChange([])
+    }
+    const selectedDataSet = dataSets.find(ds => ds.id === selectedDataSetId)
+
+    return (
+        <>
+            <div className={styles.selectsContainer}>
+                <SingleSelectField
+                    className={styles.dataSetSelect}
+                    label={i18n.t('Data set')}
+                    placeholder={i18n.t('Select a data set')}
+                    selected={selectedDataSetId}
+                    filterable
+                    onChange={handleDataSetChange}
+                    dataTest="add-lock-exception-select-data-set"
+                >
+                    {dataSets.map(ds => (
+                        <SingleSelectOption
+                            key={ds.id}
+                            label={ds.displayName}
+                            value={ds.id}
+                        />
+                    ))}
+                </SingleSelectField>
+                {selectedDataSetId && (
+                    <PeriodPicker
+                        periodType={selectedDataSet.periodType}
+                        onPickPeriod={onSelectedPeriodIdChange}
+                    />
+                )}
+            </div>
+            {selectedDataSetId && (
+                <OrganisationUnitSelection
```

Comment:
The `OrgUnitSelect` components from `d2-ui` (`OrgUnitSelectByLevel`, `OrgUnitSelectByGroup`, `OrgUnitSelectAll`) from [the previous version of `AddLockExceptionForm`](https://github.com/dhis2/data-administration-app/blob/fd85975586ad812d67f88005a9db069d82b84d66/src/pages/lock-exception/AddLockExceptionForm.js) are now abstracted behind `OrganisationUnitSelection`.